src/plugins/kibana_react/public/saved_objects/saved_object_finder.tsx

```diff
@@ -174,7 +181,7 @@ class SavedObjectFinder extends React.Component<SavedObjectFinderProps, SavedObj
     }
   }, 300);
 
-  constructor(props: SavedObjectFinderProps) {
+  constructor(props: SavedObjectFinderUiProps) {
```

Comment:
Why was introducing this wrapper component necessary, instead of just introducing two new props on the existing `SavedObjectFinder` component? Are there still places where `SavedObjectFinder` is used, without `uiSettings` and `savedObjects` props?

Reply:
Not at the moment, but this is the target architecture we are moving towards (see also https://github.com/elastic/kibana/issues/52493 ). Passing `uiSettings` and `savedObjects` services down through the component tree can be cumbersome in some cases, so the wrapper component is able to pick it out of the Kibana context: https://github.com/elastic/kibana/tree/master/src/plugins/kibana_react#context

The usages of saved object finder in embeddables will move into this direction soon. Future uses should be guided into this direction.